sftp/sftp.go

```diff
@@ -0,0 +1,97 @@
+package sftp
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	// PasswordAuth indicates password-based authentication
+	PasswordAuth = "passwordAuth"
+	// KeyAuth indicates key-based authentication
+	KeyAuth = "keyAuth"
+)
+
+// SFTPManager interface abstracts the SFTP client
+type SFTPManager interface {
+	UploadFile(localFilePath, remoteDir string) error
+	DownloadFile(remoteFilePath, localDir string) error
+	DeleteFile(remoteFilePath string) error
+}
+
+// SFTPManagerImpl is a real implementation of SFTPManager
+type SFTPManagerImpl struct {
+	client SFTPClient
+}
+
+func NewSFTPManager(sshClient *ssh.Client) (SFTPManager, error) {
```

Comment:
We can probably remove `sftp` from here as well.
```suggestion
func NewManager(sshClient *ssh.Client) (SFTPManager, error) {
```